contentcuration/contentcuration/frontend/shared/views/ImmersiveModalLayout.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+
+  <VApp>
+    <VToolbar app dark color="primary">
+      <VBtn v-if="backButton" flat link :to="previousPage">
+        <Icon>arrow_back</Icon>
+        <span class="back-text">{{ appBarText }}</span>
+      </VBtn>
+
+      <template v-else>
+        <VBtn link icon dark :to="previousPage">
+          <Icon>close</Icon>
+        </VBtn>
+        <VToolbarTitle>
+          {{ appBarText }}
+        </VToolbarTitle>
+      </template>
+    </VToolbar>
+    <VContent>
+      <div class="content-wrapper">
+        <PageContainer class="content">
+          <slot></slot>
+        </PageContainer>
+      </div>
+    </VContent>
+  </VApp>
+
+</template>
+
+
+<script>
+
+  import PageContainer from './PageContainer';
+
+  export default {
+    name: 'ImmersiveModalLayout',
+    components: {
+      PageContainer,
+    },
+    props: {
+      previousPage: {
+        type: [String, Object],
+        required: true,
+      },
+      backButton: {
+        type: Boolean,
+        default: false,
+      },
+      appBarText: {
+        type: String,
+        required: true,
+      },
+    },
+  };
+
+</script>
+
+
+
+<style lang="less" scoped>
+
+  .back-text {
+    display: inline-block;
+    margin-left: 8px;
+    font-size: 16px;
+    text-transform: initial;
+  }
+
+  .content-wrapper {
+    margin: 32px;
+  }
+
+  .content {
+    width: 500px;
```

Comment:
Should this be min-width?

Reply:
no, that breaks on mobile. It needs to be able to squish down

Reply:
what behavior are you seeing?

Reply:
If I want to use this for a page that might require more space like the edit modal, want to avoid limiting it to 500px

Reply:
cool - I'll remove that line for now